src/valueTypes/StringValueType/component.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import theme from '../../utils/theme'
+import useUiEventListener from '../../utils/customReactHooks/useUiEventListener'
+import { useStore } from 'react-redux'
+
+const TextField = styled.input`
+    height: 17px;
+    border: 2px solid black;
+    border-radius: 5px;
+    background: ${props => props.theme === 'panel' ? theme.bgColorDark3 : theme.bgColorDark2};
+    color:${theme.textColorLight1};
+    cursor: text;
+    width: 100%;
+`
+
+const ParamString = ({ onChange, nodeId }) => {
+  const store = useStore()
+  const [value, setValue] = useState()
+
+  useUiEventListener('slow-tick', () => {
+    const newValue = store.getState().nodes[nodeId].value
+    if (value !== newValue) setValue(newValue)
+  })
+  return (
+    <TextField
+      value={value || ''}   //This seems to not handle multiple fast keypresses, only updating with one of the keys
```

Comment:
Is this the correct way to be feeding the value into the component? as the comment states, it works well enough until you start typing fast

Reply:
I just had the thought that perhaps why it wasn't handling fast typing is because it was updating the geometry with each change, which could have cause slight lag/frame-drops (especially as your text gets longer)

Reply:
I think we might want to have some sort of debounce option for the text field. We kind of have this with the `slow-tick` event listener though. My advice would be to get it working with canvas on a plane first, before going for full 3D text.

Reply:
I actually believe the 3d lag can be mitigated by having a sketch that generates all the character geometry on creation then re-assembles into a parent object container as the text updates. Having a debounce, or not applying the text until enter is pressed could be good though